src/components/EnsureDeviceApp/index.js

```diff
@@ -116,31 +117,42 @@ class EnsureDeviceApp extends PureComponent<Props, State> {
     let isSuccess = true
 
     try {
-      if (account) {
+      if (account || currency) {
+        const cur = account ? account.currency : currency
+        invariant(cur, 'currency is available')
         const { address } = await getAddress
           .send({
             devicePath: deviceSelected.path,
-            currencyId: account.currency.id,
-            path: account.freshAddressPath,
-            segwit: isSegwitAccount(account),
+            currencyId: cur.id,
+            path: account
+              ? account.freshAddressPath
+              : standardDerivation({ currency: cur, segwit: false, x: 0 }),
+            segwit: account ? isSegwitAccount(account) : false,
           })
           .toPromise()
-        const { freshAddress } = account
-        if (account && freshAddress !== address) {
-          logger.warn({ freshAddress, address })
-          throw new Error('Account address is different than device address')
-        }
-      } else if (currency) {
-        const pass = await isCurrencyAppOpened
-          .send({
-            devicePath: deviceSelected.path,
-            currencyId: currency.id,
+          .catch(e => {
+            if (
+              e &&
+              (e.name === 'TransportStatusError' ||
+                // we don't want these error to appear (caused by usb disconnect..)
+                e.message === 'could not read from HID device' ||
+                e.message === 'Cannot write to HID device')
+            ) {
+              logger.log(e)
+              throw new Error(`You must open application ‘${cur.name}’ on the device`)
```

Comment:
is it intended to use french quotes here?

Reply:
not really. what do you think we should use?

Reply:
““ maybe ?

Reply:
''
""
‘‘
““

Reply:
After all, let's keep it like that. It's pretty elegant :nail_care: